pkg/apisix/resource_test.go

```diff
@@ -74,3 +75,18 @@ func TestItemConvertRoute(t *testing.T) {
 	assert.Equal(t, r.Methods[1], "POST")
 	assert.Equal(t, r.Name, "unknown")
 }
+
+func TestRouteVarsUnmarshalJSONCompatibility(t *testing.T) {
+	var route v1.Route
+	data := `{"vars":{}}`
+	err := json.Unmarshal([]byte(data), &route)
+	assert.Nil(t, err)
+
+	data = `{"vars":{"a":"b"}}`
+	err = json.Unmarshal([]byte(data), &route)
+	assert.Equal(t, err.Error(), "unexpected non-empty object")
+
+	data = `{"vars":[]}`
+	err = json.Unmarshal([]byte(data), &route)
+	assert.Nil(t, err)
```

Comment:
Also add a normal case where `vars` is a non-empty slice.

Reply:
The test case has been added!